static/js/services/screensharing.js

```diff
@@ -137,40 +177,29 @@ define(['underscore', 'text!partials/screensharedialogff.html', 'webrtc.adapter'
 
 				// Firefox 36 got screen sharing support.
 				// See https://bugzilla.mozilla.org/show_bug.cgi?id=923225
-				if ($window.webrtcDetectedVersion >= 36) {
+				if ($window.webrtcDetectedVersion >= 36 && firefoxExtension.available) {
 					this.supported = true;
 					this.prepare = function(options) {
-						// To work, the current domain must be whitelisted in
-						// media.getusermedia.screensharing.allowed_domains (about:config).
-						// See https://wiki.mozilla.org/Screensharing for reference.
-						var d = $q.defer();
-						var dlg = dialogs.create('/dialogs/screensharedialogff.html', screenshareDialogFFController, {selection: "screen"}, {});
-						dlg.result.then(function(source) {
-							if (source) {
-								var opts = _.extend({
-									mediaSource: source
-								}, options);
-								d.resolve(opts);
-							} else {
-								d.resolve(null);
-							}
-						}, function(err) {
-							d.resolve(null);
-						});
-						return d.promise;
+						return selectFirefoxScreenToShare(options);
 					};
+				} else {
+
+					if (firefoxExtension.autoinstall.force) {
+						this.supported = false;
+					}
+
 				}
 
 			} else {
 				// No support for screen sharing.
 			}
 
+			var waiting = false;
+			var prepareAlternative = this.prepare;
```

Comment:
What is not good about this? The variables are available in this scope level anyway. Declaring them here makes it more clear when used for Firefox as well.